pytorch_lightning/accelerators/registry.py

```diff
@@ -0,0 +1,121 @@
+    to the accelerator flag to the Trainer.
+
+    Example::
+
+        @AcceleratorRegistry.register("sota", description="Custom sota accelerator", a=1, b=True)
+        class SOTAAccelerator(Accelerator):
+            def __init__(self, a, b):
+                ...
+
+        or
+
+        AcceleratorRegistry.register("sota", SOTAAccelerator, description="Custom sota accelerator", a=1, b=True)
+    """
+
+    def register(
+        self,
+        name: str,
+        accelerator: Optional[Callable] = None,
+        description: str = "",
+        override: bool = False,
+        **init_params: Any,
+    ) -> Callable:
+        """Registers a accelerator mapped to a name and with required metadata.
+
+        Args:
+            name : the name that identifies a accelerator, e.g. "gpu"
+            accelerator : accelerator class
+            description : accelerator description
+            override : overrides the registered accelerator, if True
+            init_params: parameters to initialize the accelerator
+        """
+        if not (name is None or isinstance(name, str)):
+            raise TypeError(f"`name` must be a str, found {name}")
+
+        if name in self and not override:
+            raise MisconfigurationException(f"'{name}' is already present in the registry. HINT: Use `override=True`.")
+
+        data: Dict[str, Any] = {}
+
+        data["init_params"] = init_params
+
+        def do_register(name: str, accelerator: Callable) -> Callable:
+            data["accelerator"] = accelerator
+            data["accelerator_name"] = name
+            self[name] = data
+            return accelerator
+
+        if accelerator is not None:
+            return do_register(name, accelerator)
+
+        return do_register
+
+    def get(self, name: str, default: Optional[Any] = None) -> Any:
+        """Calls the registered accelerator with the required parameters and returns the accelerator object.
+
+        Args:
+            name (str): the name that identifies a accelerator, e.g. "gpu"
+        """
+        if name in self:
+            data = self[name]
+            return data["accelerator"](**data["init_params"])
+
+        if default is not None:
+            return default
+
+        err_msg = "'{}' not found in registry. Available names: {}"
+        available_names = self.available_accelerators()
+        raise KeyError(err_msg.format(name, available_names))
+
+    def remove(self, name: str) -> None:
+        """Removes the registered accelerator by name."""
+        self.pop(name)
+
+    def available_accelerators(self) -> List[str]:
+        """Returns a list of registered accelerators."""
+        return list(self.keys())
+
+    def __str__(self) -> str:
+        return "Registered Accelerators: {}".format(", ".join(self.available_accelerators()))
+
+
+AcceleratorRegistry = _AcceleratorRegistry()
+
+
+def call_register_accelerators(base_module: str) -> None:
```

Comment:
maybe merge this with https://github.com/PyTorchLightning/pytorch-lightning/blob/7a6efb38b25d6d4de75a09199809f70e0b930488/pytorch_lightning/strategies/strategy_registry.py#L133 so that we have a single function taking the base_module, the registry and the type to register?

No need to fully duplicate this. We could also think of an abstract registry class, having this as a function so that we spare the registry argument (and get `self` instead).

Reply:
Could take a stab at it with a follow up PR, need the registry changes for Habana PR now